ts/main.ts

```diff
@@ -30,6 +30,90 @@ $form?.addEventListener('submit', (event: Event) => {
   data.nextEntryId++;
   data.entries.unshift(obj);
 
+  const newList = renderEntry(obj);
+
+  const theFirstChild = $ul?.firstChild as HTMLElement;
+  $ul?.insertBefore(newList, theFirstChild);
+  viewSwap('entries');
+  toggleEntries();
+
   $img.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form.reset();
 });
+
+function renderEntry(entry: Obj): HTMLElement {
+  const $entriesList = document.createElement('li');
+  $entriesList.className = 'entries-list row';
+
+  const $divImg = document.createElement('div');
+  $divImg.className = 'column-half';
+  $entriesList.appendChild($divImg);
+
+  const $imgList = document.createElement('img');
+  $imgList.setAttribute('src', entry.photoURL);
+  $divImg.appendChild($imgList);
+
+  const $divList = document.createElement('div');
+  $divList.className = 'column-half';
+  $entriesList.appendChild($divList);
+
+  const $h2List = document.createElement('h2');
+  $h2List.textContent = entry.title;
+  $divList.appendChild($h2List);
+
+  const $pList = document.createElement('p');
+  $pList.textContent = entry.notes;
+  $divList.appendChild($pList);
+
+  return $entriesList;
+}
+
+const $ul = document.querySelector('.entries-ul');
+
+document.addEventListener('DOMContentLoaded', () => {
+  for (let i = 0; i < data.entries.length; i++) {
+    const currentEntry = renderEntry(data.entries[i]);
+    $ul?.appendChild(currentEntry);
+    const previousView = data.view;
+    viewSwap(previousView);
+    toggleEntries();
```

Comment:
These should be outside the loop, otherwise you will be viewSwapping for every single item in the array